### PR TITLE
OSX and profile testing compatibility

### PR DESCRIPTION
**Pull request by alexpott**
This PR allows it to be used to run profile tests and fixes a few issues on OSX related to its use of bash version 3.

### CODE REVIEW

File: bin/test-drupal-module [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+trap exit 1 ERR
```
Comment (dbosen) [inline bin/test-drupal-module]: I tested a bit with this, but it seems set -e is more reliable, not sure why though.

File: includes/testing-library.sh [hunk 2]
```diff
@@ -120,6 +155,8 @@ clean_up() {
     fi
 
     docker rm -f -v selenium-for-tests
+    # Remove any subprocesses. For example the drush webserver process
+    [[ -z "$(jobs -p)" ]] || kill $(jobs -p)
```
Comment (dbosen) [inline includes/testing-library.sh]: This will not work if the script is not run in one go.

File: includes/testing-library.sh [hunk 3]
```diff
@@ -120,6 +155,8 @@ clean_up() {
     fi
 
     docker rm -f -v selenium-for-tests
+    # Remove any subprocesses. For example the drush webserver process
+    pkill -P $$
```
Comment (dbosen) [inline includes/testing-library.sh]: This will not work if the script is not run in one go

File: includes/testing-library.sh [hunk 4]
```diff
@@ -276,20 +313,27 @@ _stage_run_tests() {
...
+            fi
+            php ${phpunit} --verbose --debug --configuration ${docroot}/core ${test_selection} ${project_test_directory} || exit 1
         ;;
         "run-tests")
-            php ${runtests} --php $(which php) --suppress-deprecations --verbose --color --url http://${DRUPAL_TRAVIS_HTTP_HOST}:${DRUPAL_TRAVIS_HTTP_PORT} ${DRUPAL_TRAVIS_TEST_GROUP} || exit 1
+            if [ ${DRUPAL_TRAVIS_TEST_GROUP} ]; then
+               test_selection="${DRUPAL_TRAVIS_TEST_GROUP}"
+            else
+               test_selection="--directory ${project_test_directory}"
+            fi
```
Comment (dbosen) [inline includes/testing-library.sh:332]: Why is this mutual exclusive?

File: includes/testing-library.sh [hunk 5]
```diff
@@ -86,6 +99,28 @@ get_composer_bin_dir() {
     echo ${composer_bin_dir}
 }
 
+get_project_type_directory() {
```
Comment (dbosen) [inline includes/testing-library.sh:102]: We now have one function suffixed with "_dir" and one with "_directory", we should keep that consistent. I like your version better, so let's rename get_composer_bin_dir to get_composer_bin_directory.

**Review by dbosen — CHANGES_REQUESTED** [review]

File: bin/test-drupal-module (hunk 1 above)
Comment (alexpott) [reply bin/test-drupal-module]: Okay let's swap it. I think failing on error is important.

File: includes/testing-library.sh (hunk 3 above)
Comment (alexpott) [reply includes/testing-library.sh]: Let's handle this in a separate issue then.

File: includes/testing-library.sh (hunk 4 above)
Comment (alexpott) [reply includes/testing-library.sh:332]: Because I don't trust run-tests.sh to do anything sensible.

Comment (alexpott) [reply includes/testing-library.sh:332]: And my mistrust is well placed. The options are exclusive.

Comment (dbosen) [reply includes/testing-library.sh:332]: Wonderful :)

**Review by dbosen — APPROVED** [review]

**Comment by dbosen** [comment]
Thank you very much!

**Pull request merged**